### PR TITLE
[release/5.0] Update API baseline files

### PR DESCRIPTION
**Pull request by dougbu**
- #26784
- move all PublicAPI.Unshipped.txt files to PublicAPI.Shipped.txt
- add eng/PublicAPI.empty.txt file
- add missing public API baseline files
- reference Microsoft.CodeAnalysis.PublicApiAnalyzers more often

### CODE REVIEW

**Comment by dougbu** [comment]
This is pure infrastructure and therefore a tell-mode change. But, it's blocked until release/5.0 branch is rebranded for 5.0.1.

**Comment by dougbu** [comment]
Reviewers: I suggest ignoring all `*.txt` files when looking at the differences. `git` and the GitHub UI don't seem to be able to handle the many renames.

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X.csproj [hunk 1]
```diff
@@ -9,6 +9,8 @@
     <IsShipping>false</IsShipping>
     <!-- Need to build this project in source build -->
     <ExcludeFromSourceBuild>false</ExcludeFromSourceBuild>
+    <!-- No need to track public APIs of this design-time build infrastructure. -->
+    <AddPublicApiAnalyzers>false</AddPublicApiAnalyzers>
```
Comment (dougbu) [inline src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X.csproj:13]: @JunTaoLuo @pranavkm do you agree with this assessment❔ Given the few references to this project and Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X.csproj, it's not clear to me how they're used.

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj [hunk 2]
```diff
@@ -7,14 +7,16 @@
...
 
     <!-- Allow assemblies outside of lib in the package -->
     <NoWarn>$(NoWarn);NU5100</NoWarn>
     <!-- It is intentional to not include build/netstandard2.0/Microsoft.NET.Sdk.Razor.targets -->
     <NoWarn>$(NoWarn);NU5129</NoWarn>
     <!-- Need to build this project in source build -->
     <ExcludeFromSourceBuild>false</ExcludeFromSourceBuild>
+    <!-- Do not disable ApiCheck. We need to be aware of when we make changes to the Tasks since they need to be backwards compatible. -->
+    <!-- No need to track public APIs of these MSBuild tasks. -->
+    <AddPublicApiAnalyzers>false</AddPublicApiAnalyzers>
```
Comment (dougbu) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:19]: @pranavkm pick one 😺

**Comment by dougbu** [comment]
Looking into the Microsoft.AspNetCore.Components.Authorization failures. Seems likely the analyzer wasn't run at the right time in VS or that it's still run twice. You fixed the second problem in release/5.0 didn't you @pranavkm❔

File: src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Compilation.targets [hunk 3]
```diff
@@ -162,6 +162,7 @@ Copyright (c) .NET Foundation. All rights reserved.
          ResponseFiles="$(CompilerResponseFile)"
          RuntimeMetadataVersion="$(RuntimeMetadataVersion)"
          SharedCompilationId="$(SharedCompilationId)"
+         SkipAnalyzers="true"
```
Comment (dougbu) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Compilation.targets:165]: @ajaybhargavb @NTaylorMullen @pranavkm is there a downside here (if it solves the CI problems)❔

**Comment by dougbu** [comment]
Ping reviewers (tomorrow): This seems to be working fine now and I've got a lot of open questions&hellip;

**Comment by dougbu** [comment]
Validation build broken for the same reason as the release/5.0 branch. But, please answer the questions above while we're working to solve that general problem. 

Put another way, ping reviewers 😺

**Comment by dougbu** [comment]
Removed the blocking labels Still need help from reviewers❕

@pranavkm I see I directed most of my questions to you. If you're not the right person to answer some or all of them, please redirect.

File: src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X.csproj (hunk 1 above)
Comment (pranavkm) [reply src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X.csproj:13]: Yup. This isn't required. The assembly is only referenced by the SDK / razor compiler and does not get shipped.

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj (hunk 2 above)
Comment (pranavkm) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:19]: Let's go with "No need to track public APIs of these MSBuild tasks". We used to have scenarios where 2.2 MSBuild targets referenced tasks from the SDK, but we broke that at some point. So the compat question is moot.

File: src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Compilation.targets (hunk 3 above)
Comment (pranavkm) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Compilation.targets:165]: This one's problematic. For instance, the "don't use HtmlHelper.Partial" analyzer that we wrote needs to run here (among other analyzers). We might also need analyzers once we start using source generators for Razor. 

Is there a reason to do this?

Comment (dougbu) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Compilation.targets:165]: I introduced this in https://github.com/dotnet/aspnetcore/pull/27653/commits/97bd5d5dd4309edee81c4e0c5b150f5d69c68291 The PublicAPI analyzer was still running in the pre-build for *.cshtml` and `*.razor` files. That meant the analyzer failed the build due to missing `public` methods et cetera. If this is a problem, I could spend time looking for a way to disable just that analyzer and just for this task.

**Review by javiercn — APPROVED** [review]
⚰️

File: Directory.Build.props [hunk 4]
```diff
@@ -11,14 +11,23 @@
...
     <IsUnitTestProject Condition=" '$(IsUnitTestProject)' == '' ">false</IsUnitTestProject>
-    <IsUnitTestProject Condition="'$(IsSpecificationTestProject)' != 'true' and ( $(MSBuildProjectName.EndsWith('Tests')) or $(MSBuildProjectName.EndsWith('.Test')) or $(MSBuildProjectName.EndsWith('.FunctionalTest')) )">true</IsUnitTestProject>
-    <IsTestAssetProject Condition="$(RepoRelativeProjectDir.Contains('testassets'))">true</IsTestAssetProject>
-    <IsSampleProject Condition="$(RepoRelativeProjectDir.Contains('sample'))">true</IsSampleProject>
+    <IsUnitTestProject Condition=" '$(IsSpecificationTestProject)' != 'true' AND
+        ( $(MSBuildProjectName.EndsWith('Tests')) OR
+          $(MSBuildProjectName.EndsWith('.Test')) OR
+          $(MSBuildProjectName.EndsWith('.FunctionalTest')) ) ">true</IsUnitTestProject>
+    <IsTestAssetProject Condition=" $(RepoRelativeProjectDir.Contains('testassets')) ">true</IsTestAssetProject>
+    <IsSampleProject Condition=" $(RepoRelativeProjectDir.Contains('sample')) OR
```
Comment (wtgodbe) [inline Directory.Build.props:24]: Could probably use `.ToLower` here

Comment (dougbu) [reply Directory.Build.props:24]: Hmm, you're right but that's not worth another iteration. I'll include the suggestion in another PR at some point.

**Review by wtgodbe — APPROVED** [review]
LGTM other than the one nit

File: Directory.Build.props (hunk 4 above)
Comment (dougbu) [reply Directory.Build.props:24]: Added this fixup to #27473

**Pull request merged**